src/OrleansProviders/Streams/Memory/MemoryBatchContainer.cs

```diff
@@ -9,10 +7,10 @@
 
 namespace Orleans.Providers
 {
-    [Serializable]
     internal class MemoryBatchContainer<TSerializer> : IBatchContainer
-        where TSerializer : IMemoryMessageBodySerializer, new()
+        where TSerializer : IMemoryMessageBodySerializer
     {
+        private readonly IMemoryMessageBodySerializer serializer;
```

Comment:
hmmm, this can be a little tricky... I missed that... supplying your own serializer might not be straightforward. Still I like it a little bit better than supplying SM on every call, but in that case not sure if I like it better than supplying IServiceProvider on every call, or at the very least have an explicit interface for binding it after deserializing (bind to an SP, not to SM). it can be that this interface is optional, and the binding call happens only if the serializer implements it